java/ql/lib/semmle/code/java/Statement.qll

```diff
@@ -557,13 +557,7 @@ class ThrowStmt extends Stmt, @throwstmt {
 }
 
```

Comment:
We generally try to avoid exposing public `abstract` classes like this, since subclassing using `extends` can lead to surprising results. It is better to define the union type explicitly:
```suggestion

private class JumpStmt_ = @breakstmt or @yieldstmt or @continuestmt;

```

Reply:
Good point, thanks! I have implemented your suggestion manually, and added a small comment explaining the rationale; hopefully that is alright.

A bit unrelated, but I have created a https://github.com/github/codeql/discussions/8713 to discuss making implementing abstract classes less error-prone.